File: leetcode/5-Longest-Substring-Without-Repeating-Characters/Longest-Substring-Without-Repeating-Characters.cpp
//Runtime: 20 ms, faster than 87.81% of C++ online submissions for Longest Substring Without Repeating Characters.
//Memory Usage: 13.9 MB, less than 98.44% of C++ online submissions for Longest Substring Without Repeating Characters.
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        	int max = 0;
	string subS = "";
	int leftIndex = 0;
	for (int i = 0; i < s.length(); i++) {
		int repeat = subS.find(s[i]);
		if (repeat != -1) {
			leftIndex = leftIndex + repeat + 1;
		}
		subS = s.substr(leftIndex, i + 1 - leftIndex);
		max = subS.length() > max? subS.length():max;
	}
	return max;
    }
};

File: leetcode/5-Longest-Substring-Without-Repeating-Characters/Longest-Substring-Without-Repeating-Characters.go
//Runtime: 4 ms, faster than 100.00% of Go online submissions for Longest Substring Without Repeating Characters.
//Memory Usage: 2.6 MB, less than 90.46% of Go online submissions for Longest Substring Without Repeating Characters.
func lengthOfLongestSubstring(s string) int {
    	max := 0
	subS := ""
	leftIndex := 0
	for i:=0;i< len(s);i++{
		repeat := strings.IndexAny(subS, s[i:i+1])
		if repeat != -1{
			leftIndex = leftIndex + repeat + 1
		}
		subS = s[leftIndex: i + 1]
		if len(subS) > max {
			max = len(subS)
		}
	}
	return max
}